packages/material-ui-styles/src/makeStyles/makeStyles.js

```diff
@@ -140,14 +140,16 @@ function detach({ state, theme, stylesOptions, stylesCreator }) {
   }
 
   const sheetManager = multiKeyStore.get(stylesOptions.sheetsManager, stylesCreator, theme);
-  sheetManager.refs -= 1;
-  const sheetsRegistry = stylesOptions.sheetsRegistry;
-
-  if (sheetManager.refs === 0) {
-    multiKeyStore.delete(stylesOptions.sheetsManager, stylesCreator, theme);
-    stylesOptions.jss.removeStyleSheet(sheetManager.staticSheet);
-    if (sheetsRegistry) {
-      sheetsRegistry.remove(sheetManager.staticSheet);
+  if (sheetManager) {
```

Comment:
While this might prevent the crash it also might hide an actual defect. It may be the right fix but we need a test to understand the issue.